### PR TITLE
Fix test_implicit_cast_ISO_formats_with_time_zone

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

Need to account for daylight savings

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/types/TimeTZTypeTest.java [hunk 1]
```diff
@@ -36,7 +37,10 @@ public void test_implicit_cast_null() {
 
     @Test
     public void test_implicit_cast_ISO_formats_with_time_zone() {
-        assertThat(TimeTZType.INSTANCE.implicitCast("01:00:00     CET"), isTZ(3600000000L, 7200));
+        assertThat(
+            TimeTZType.INSTANCE.implicitCast("01:00:00     CET"),
+            Matchers.anyOf(isTZ(3600000000L, 3600), isTZ(3600000000L, 7200))
```
Comment (seut) [inline server/src/test/java/io/crate/types/TimeTZTypeTest.java:42]: Should it always result in `isTZ(3600000000L, 3600)` for timezone `CET`?

File: docs/general/ddl/data-types.rst [hunk 2]
```diff
@@ -487,15 +487,7 @@ Time zone syntax as defined by `ISO 8601 time zone designators`_.
     +------------------+
     SELECT 1 row in set (... sec)
 
-    cr> select '13:59:59.999999 CET'::timetz;
-    +-----------------------+
-    | 13:59:59.999999+02:00 |
```
Comment (seut) [inline docs/general/ddl/data-types.rst:492]: Afaik this is wrong and should be fixed inside the implementation.
PostgreSQL:
```
postgres=# select '13:59:59.999999 CET'::timetz;
       timetz       
--------------------
 13:59:59.999999+01
(1 row)

postgres=# select '13:59:59.999999 CEST'::timetz;
       timetz       
--------------------
 13:59:59.999999+02
(1 row)
```

Comment (mfussenegger) [reply docs/general/ddl/data-types.rst:492]: The same result is now also returned by CrateDB:

```
crate=> select '13:59:59.999999 CET'::timetz;
 13:59:59.999999+01:00
-----------------------
 13:59:59.999999+01:00
(1 row)
```


The problem is that the output is in relation to UTC, and depending on the
daylight savings CET can be +2 or +1 hour.

Comment (mfussenegger) [reply docs/general/ddl/data-types.rst:492]: It is indeed a bit confusing. From https://time.is/nl/CET

> CET alternates between UTC+1 (standard time) and UTC+2 (when daylight saving time (DST) is observed). All countries in the CET time zone observe DST (UTC+2) from 02:00 am on the last Sunday of March until 03:00 am on the last Sunday of October.

> Warning: Many sources define CET as a constant UTC+1. In common usage however, CET usually refers to the time observed in most of Europe, be it standard time or daylight saving time.

Comment (seut) [reply docs/general/ddl/data-types.rst:492]: Hm I don't understand this as `CET` is always `+1` and never `+2` and opposite for `CEST` which always `+2`. What do I miss? ;)

**Review by seut — APPROVED** [review]

**Comment by seut** [comment]
retest this please
Some unrelated failures occurred:
```
REPRODUCE WITH: ./gradlew null -Dtests.seed=2416B2A1506B8DD -Dtests.class=io.crate.integrationtests.ResizeShardsITest -Dtests.method="testShrinkShardsEnsureLeftOverIndicesAreRemoved" -Dtests.locale=fr-LU -Dtests.timezone=Australia/Melbourne
REPRODUCE WITH: ./gradlew null -Dtests.seed=2416B2A1506B8DD -Dtests.class=io.crate.integrationtests.ResizeShardsITest -Dtests.method="testShrinkShardsOfPartition" -Dtests.locale=fr-LU -Dtests.timezone=Australia/Melbourne

java.lang.IllegalStateException: index csdaqwf.quotes must have all shards allocated on the same node to shrink index
	at __randomizedtesting.SeedInfo.seed([2416B2A1506B8DD:9440255F226393E7]:0)
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.validateShrinkIndex(MetadataCreateIndexService.java:618)
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.prepareResizeIndexSettings(MetadataCreateIndexService.java:672)
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService$IndexCreationTask.execute(MetadataCreateIndexService.java:403)
	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45)
	at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:690)
	at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:308)
	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:208)
	at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:140)
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:153)
	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:191)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:259)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:222)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	
```

**Pull request merged**